File: app/src/main/java/org/stepic/droid/ui/custom_exo/AppCompatSeekTimeBar.java
package org.stepic.droid.ui.custom_exo;

import android.content.Context;
import android.support.annotation.Nullable;
import android.support.v7.widget.AppCompatSeekBar;
import android.util.AttributeSet;

public class AppCompatSeekTimeBar extends AppCompatSeekBar implements TimeBar {

    private TimeBar.OnScrubListener listener;

    public AppCompatSeekTimeBar(Context context) {
        super(context);
    }

    public AppCompatSeekTimeBar(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public AppCompatSeekTimeBar(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public void setListener(OnScrubListener listener) {
        this.listener = listener;
    }

    @Override
    public void setKeyTimeIncrement(long time) {

    }

    @Override
    public void setKeyCountIncrement(int count) {

    }

    @Override
    public void setPosition(long position) {
        setProgress((int) position);
    }

    @Override
    public void setBufferedPosition(long bufferedPosition) {
        setSecondaryProgress((int) bufferedPosition);
    }

    @Override
    public void setDuration(long duration) {
        setMax((int) duration);
    }

    @Override
    public void setAdBreakTimesMs(@Nullable long[] adBreakTimesMs, int adBreakCount) {
        //// TODO: 10.05.17 implement it, when needed
    }
}

File: app/src/main/java/org/stepic/droid/ui/custom_exo/AppCompatSeekTimeBar.kt
package org.stepic.droid.ui.custom_exo

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import android.view.View
import android.widget.SeekBar
import androidx.appcompat.widget.AppCompatSeekBar
import com.google.android.exoplayer2.ui.TimeBar
import timber.log.Timber

class AppCompatSeekTimeBar
@JvmOverloads
constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0): AppCompatSeekBar(context, attrs, defStyleAttr), TimeBar, SeekBar.OnSeekBarChangeListener {
    private val listeners: MutableList<TimeBar.OnScrubListener> = mutableListOf()
    private val notTouchableOnTouchListener: (View?, MotionEvent?) -> Boolean by lazy {
        { _: View?, _: MotionEvent? -> true }
    }

    init {
        setOnSeekBarChangeListener(this)
    }

    override fun addListener(listener: TimeBar.OnScrubListener?) {
        listener?.let(listeners::add)
    }

    override fun removeListener(listener: TimeBar.OnScrubListener?) {
        listener?.let(listeners::remove)
    }

    override fun setKeyTimeIncrement(time: Long) {
        Timber.d("setKeyTimeIncrement($time)")
    }

    override fun setKeyCountIncrement(count: Int) {
        Timber.d("setKeyCountIncrement($count)")
    }

    override fun setPosition(position: Long) {
        progress = position.toInt()
    }

    override fun setBufferedPosition(bufferedPosition: Long) {
        secondaryProgress = bufferedPosition.toInt()
    }

    override fun setDuration(duration: Long) {
        max = duration.toInt()
    }

    override fun setAdGroupTimesMs(adGroupTimesMs: LongArray?, playedAdGroups: BooleanArray?, adGroupCount: Int) {
        // TODO: 10.05.17 implement it, when needed
    }

    override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
        if (fromUser) {
            listeners.forEach {
                it.onScrubMove(this, progress.toLong())
            }
        }
    }

    override fun onStartTrackingTouch(seekBar: SeekBar?) {
        listeners.forEach {
            it.onScrubStart(this, progress.toLong())
        }
    }

    override fun onStopTrackingTouch(seekBar: SeekBar?) {
        listeners.forEach {
            it.onScrubStop(this, progress.toLong(), false)
        }
    }

    override fun setEnabled(enabled: Boolean) {
        //we do it, because setEnabled add some extra effects, we should make it just not seekable
        if (!enabled) {
            setOnTouchListener(notTouchableOnTouchListener)
        } else {
            setOnTouchListener(null)
        }
    }


}
